libraries/apollo-api/src/commonMain/kotlin/com/apollographql/apollo3/api/http/internal/UrlEncode.kt

```diff
@@ -3,19 +3,16 @@ package com.apollographql.apollo3.api.http.internal
 import kotlin.native.concurrent.SharedImmutable
 
 @SharedImmutable
-private val RESERVED_CHARS = "!#\$&'\"()*+,/:;=?@[]{}"
+private val RESERVED_CHARS = "!#\$&'\"()*+,/:;=?@[]{} "
 
 /**
  * A very simple urlEncode
  */
-internal fun String.urlEncode(
-    spaceToPlus: Boolean = false
-): String = buildString {
-  this@urlEncode.forEach {
-    when {
-      it in RESERVED_CHARS -> append(it.percentEncode())
-      spaceToPlus && it == ' ' -> append('+')
-      else -> append(it)
+internal fun String.urlEncode(): String = buildString {
```

Comment:
Can you put a link to this PR, the stackoverflow question and the RFC so we can track this later if needed? 

Same thing in `HttpRequest.url`, can you add Javadoc mentionning that it expect a valid url as defined by https://www.rfc-editor.org/rfc/rfc3986#section-2 ?